### PR TITLE
Add custom vector fields and text fields for OpenSearch

### PR DESCRIPTION
**Pull request by naveentatikonda**
**Description**
Add custom vector field name and text field name while indexing and querying for OpenSearch

**Issues**
https://github.com/hwchase17/langchain/issues/2500

### CODE REVIEW

File: langchain/vectorstores/opensearch_vector_search.py [hunk 1]
```diff
@@ -250,13 +264,26 @@ def add_texts(
...
+
+            text_field: Document field the text of the document is stored in. Defaults
+            to "text".
         """
-        embeddings = [
-            self.embedding_function.embed_documents([text])[0] for text in texts
-        ]
+        embeddings = self.embedding_function.embed_documents(list(texts))
         _validate_embeddings_and_bulk_size(len(embeddings), bulk_size)
+        vector_field = _get_kwargs_value(kwargs, "vector_field", "vector_field")
```
Comment (vowelparrot) [inline langchain/vectorstores/opensearch_vector_search.py:277]: This is a meta-comment on a previous PR, but `_get_kwargs_value` could just be `kwargs.get(key, default_value)`

**Review by vowelparrot — APPROVED** [review]

**Review by hwchase17 — APPROVED** [review]
thanks!

File: langchain/vectorstores/opensearch_vector_search.py (hunk 1 above)
Comment (jponf) [reply langchain/vectorstores/opensearch_vector_search.py:277]: @vowelparrot there is a PR addressing this https://github.com/langchain-ai/langchain/pull/13184

**Comment by manisha1609** [comment]
how to define same using langchain javascript opensearchvectorstore?

**Pull request merged**